corehq/apps/api/urls.py

```diff
@@ -119,7 +120,8 @@ def api_url_patterns():
               ODataFormMetadataView.as_view(), name=ODataFormMetadataView.table_urlname)
     yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/\$metadata$',
               ODataFormMetadataView.as_view(), name=ODataFormMetadataView.urlname)
-    yield url(r'v0.6/case/', include(case_api_urlpatterns))
+    yield url(r'v0.6/case$', case_api)  # We don't want Django to redirect to add a trailing slash
+    yield url(r'v0.6/case/?', include(case_api_urlpatterns))
```

Comment:
This should work. It reduces duplication and allows you to drop `case_api_urlpatterns`, but it's not as simple.
```suggestion
    yield url(r'v0.6/case(?:/(?P<case_id>[\w-]+)?/?)?$', case_api, name='case_api')
```

Reply:
Alternate that does not match `v0.6/case//`:
```suggestion
    yield url(r'v0\.6/case(?:/(?P<case_id>[\w-]+))?/?$', case_api, name='case_api')
```

Reply:
Ah! Nice.

Reply:
Thanks you two, just tried it out (including with tests) and it works like a charm.  The `\.` didn't escape my notice either - nice touch.